### ISSUE TITLE
CDK update check after certain commands takes a very long time

### ISSUE THREAD

**Issue by MichaelHindley** [aws/aws-cdk#3518]
* **I'm submitting a ...**
  - [x] :beetle: bug report

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*
running `cdk list` or `cdk deploy` results in a long wait time at the end of the command before it exits.

![image](https://user-images.githubusercontent.com/1016639/62358404-f079d380-b514-11e9-8884-f1cc020e64f3.png)

In the screenshot above, there is a delay of upwards a minute between the Stack arn being printed and the "Newer version of CDK is available".

* **What is the expected behavior (or behavior of feature suggested)?**
The update check should be almost instant.

* **Please tell us about your environment:**

  - CDK CLI Version: 1.1.0
  - OS: Windows 10
  - Language: Typescript

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

**Comment by MichaelHindley** [comment]
The environment running these commands is also behind a proxy.

**Comment by shivlaks** [comment]
I tried a few different project creation / stack deploy worflows (all behind a proxy) and I can't seem to reproduce this scenario. It typically spits out the output right away. 

Is there any chance you could provide more details about the environment? running the cli commands with the `--v` flag might also provide more hints

**Comment by SomayaB** [comment]
> I tried a few different project creation / stack deploy worflows (all behind a proxy) and I can't seem to reproduce this scenario. It typically spits out the output right away.
> 
> Is there any chance you could provide more details about the environment? running the cli commands with the `--v` flag might also provide more hints

@MichaelHindley

**Comment by MichaelHindley** [comment]
Running `cdk list --v` in my local environment. The command is "done" when it encounters an error due to missing environment variable, but the process is still running:

```pre
CDK toolkit version: 1.12.0 (build 923055e)
Command line arguments: { _: [ 'list' ],
  v: true,
  verbose: true,
  'ignore-errors': false,
  ignoreErrors: false,
  json: false,
  j: false,
  ec2creds: undefined,
  i: undefined,
  'version-reporting': undefined,
  versionReporting: undefined,
  'path-metadata': true,
  pathMetadata: true,
  'asset-metadata': true,
  assetMetadata: true,
  'role-arn': undefined,
  r: undefined,
  roleArn: undefined,
  staging: true,
  'no-color': false,
  noColor: false,
  long: false,
  l: false,
  '$0':
   'C:\\Users\\MHXJPG\\AppData\\Roaming\\npm\\node_modules\\aws-cdk\\bin\\cdk' }
Determining whether we're on an EC2 instance.
Using proxy server: http://proxy:8080
cdk.json: {
  "app": "node bin/aws.js"
}
cdk.context.json: {
  "ssm:account=acc:parameterName=/aws/service/ecs/optimized-ami/amazon-linux/recommended:region=eu-west-1": "{\"schema_version\":1,\"image_name\":\"amzn-ami-2018.03.p-amazon-ecs-optimized\",\"image_id\":\"ami-0f85d8192f90863dc\",\"os\":\"Amazon Linux\",\"ecs_runtime_version\":\"Docker version 18.06.1-ce\",\"ecs_agent_version\":\"1.27.0\"}"
}
merged settings: { versionReporting: true,
  pathMetadata: true,
  output: 'cdk.out',
  app: 'node bin/aws.js',
  context: {},
  tags: [],
  assetMetadata: true,
  toolkitBucket: {},
  staging: true }
Setting "CDK_DEFAULT_REGION" environment variable to eu-west-1
Resolving default credentials
Does not look like EC2 instance.
Retrieved account ID <accountId> from disk cache
Setting "CDK_DEFAULT_ACCOUNT" environment variable to accountId
context: { 'ssm:account=accountId:parameterName=/aws/service/ecs/optimized-ami/amazon-linux/recommended:region=eu-west-1':
   '{"schema_version":1,"image_name":"amzn-ami-2018.03.p-amazon-ecs-optimized","image_id":"ami-0f85d8192f90863dc","os":"Amazon Linux","ecs_runtime_version":"Docker version 18.06.1-ce","ecs_agent_version":"1.27.0"}',
  'aws:cdk:enable-path-metadata': true,
  'aws:cdk:enable-asset-metadata': true }
outdir: cdk.out
env: { CDK_DEFAULT_REGION: 'eu-west-1',
  CDK_DEFAULT_ACCOUNT: '035746768234',
  CDK_CONTEXT_JSON:
   '{"ssm:account=accountId:parameterName=/aws/service/ecs/optimized-ami/amazon-linux/recommended:region=eu-west-1":"{\\"schema_version\\":1,\\"image_name\\":\\"amzn-ami-2018.03.p-amazon-ecs-optimized\\",\\"image_id\\":\\"ami-0f85d8192f90863dc\\",\\"os\\":\\"Amazon Linux\\",\\"ecs_runtime_version\\":\\"Docker version 18.06.1-ce\\",\\"ecs_agent_version\\":\\"1.27.0\\"}","aws:cdk:enable-path-metadata":true,"aws:cdk:enable-asset-metadata":true}',
  CDK_OUTDIR: 'cdk.out',
  CDK_CLI_ASM_VERSION: '1.10.0',
  CDK_CLI_VERSION: '1.12.0' }

C:\code\ats-aws\lib\s3-source-deployment-stack\s3-website-host.ts:96
    bucketName: `${infrastructurePrefix}-${process.env.HOSTED_ZONE_ID!.toLowerCase()}-${S3WebsiteHostConstants.sourceBucketName}`,
                                                                       ^
TypeError: Cannot read property 'toLowerCase' of undefined
    at createSourceBucket (C:\code\ats-aws\lib\s3-source-deployment-stack\s3-website-host.ts:96:72)
    at new S3WebsiteHostingBucketStack (C:\code\ats-aws\lib\s3-source-deployment-stack\s3-website-host.ts:142:27)
    at Object.<anonymous> (C:\code\ats-aws\bin\aws.ts:70:29)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:283:19)

>>>> Here it "sleeps" for 1-2 minutes <<<<
**************************************************
*** Newer version of CDK is available [1.13.1] ***
*** Upgrade recommended                        ***
**************************************************
Subprocess exited with error 1
Error: Subprocess exited with error 1
    at ChildProcess.proc.on.code (C:\Users\<user>\AppData\Roaming\npm\node_modules\aws-cdk\lib\api\cxapp\exec.ts:115:23)
    at ChildProcess.emit (events.js:189:13)
    at ChildProcess.EventEmitter.emit (domain.js:441:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)


```

The sleep occurs when there is a CDK upgrade available, and does not if there isnt. The command used doesnt seem to matter, nor if there is a error or not in the evaluated code.
